lib/deployer.rb

```diff
@@ -35,8 +35,11 @@ def deploy_all
     results = Parallel.map(
       repos,
       in_processes: Settings.num_parallel_processes,
-      finish: ->(repo, _i, _result) { progress_bar.advance(repo: repo.name) }
-    ) do |repo|
+      finish: ->(repo, _i, _result) do
+        File.open(Settings.progress_file, 'a') { |f| f.write("#{Time.now} : #{repo.name} deploy complete\n") }
```

Comment:
Because we're opening and appending to this file within `parallel`, there is a race condition possibility where multiple processes open and write to the file around the same time, which could lead to garbled output (or exceptions?). I wonder if we might want to consider:

1. Writing to a repo-specific file to avoid the race condition; and
2. Putting this behind a debug flag defaulting to "off," letting folks opt into file I/O when running into problems

❓

Reply:
Ok, logging to different filenames each time now and put a flag in that can be overriden.